src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -117,8 +117,8 @@ public boolean isPinned()
     {
         Checks.notNull(emote, "Emote");
 
-        MessageReaction reaction = reactions.parallelStream()
-                .filter(r -> Objects.equals(r.getReactionEmote().getId(), emote.getId()))
+        MessageReaction reaction = reactions.stream()
+                .filter(r -> r.getReactionEmote().getIdLong() == emote.getIdLong())
```

Comment:
This needs to filter reactions that don't have an id first, otherwise it will throw an exception.